presentation/src/main/java/com/example/presentation/navigation/CountriesNavHost.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.presentation.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import com.example.common.models.DataState
+import com.example.presentation.model.CountryUi
+import com.example.presentation.view.CountriesScreen
+import com.example.presentation.view.DetailScreen
+import com.example.presentation.viewmodel.CountriesViewModel
+import com.example.presentation.viewmodel.DetailsViewModel
+
+@Composable
+fun CountriesNavHost(navController: NavHostController) {
+    NavHost(navController = navController, startDestination = Screen.Countries.route) {
+        composable(route = Screen.Countries.route) {
+            val countriesViewModel: CountriesViewModel = hiltViewModel()
+            val uiState by countriesViewModel.uiState.observeAsState(initial = DataState.Loading)
+
+            val onItemClick: (CountryUi) -> Unit = {
+                navController.navigate(Screen.Details.route + "/" + it.id)
+            }
+
+            CountriesScreen(uiState, onItemClick)
+        }
```

Comment:
nit: to avoid having a huge NavHost when you will end up with several screens, I suggest you to create a file CountriesRoute, where you can have:
```kotlin
fun NavGraphBuilder.countriesScreen(
    onItemClick: (countryId: String) -> Unit,
) = composable(route = Screen.Countries.route) {
    CountriesRoute(onItemClick = onItemClick)
}

@Composable
fun CountriesRoute(
    onItemClick: (countryId: String) -> Unit,
    countriesViewModel: CountriesViewModel = hiltViewModel(),
) {
    val uiState by countriesViewModel.uiState.observeAsState(initial = DataState.Loading)

    CountriesScreen(
        uiState = uiState,
        onItemClick = { onItemClick(it.id) },
    )
}
```
Then update your nav host

```kotlin
@Composable
fun CountriesNavHost(navController: NavHostController) {
    NavHost(navController = navController, startDestination = Screen.Countries.route) {
        countriesScreen { countryId ->
            navController.navigate(Screen.Details.route + "/" + countryId)
        }
        // ...
    }
}
```